### ISSUE TITLE
feat(oracle): Add export-oip command with Parquet/alimentar support

### ISSUE THREAD

**Issue by noahgift** [paiml/depyler#156]
## Summary

Implement `depyler oracle export-oip` command to export CITL diagnostic corpus for OIP training pipeline integration.

## Motivation

Cross-project integration with organizational-intelligence-plugin (NLP-014) requires:
- Ground-truth defect labels from compiler diagnostics
- Scalable export format for 50K+ examples
- Integration with alimentar DataLoader

### Functional
- [ ] `depyler oracle export-oip --output <path>` command
- [ ] JSONL export format (default)
- [ ] Parquet export format (`--format parquet`)
- [ ] Error code → OIP DefectCategory mapping
- [ ] Clippy lint → OIP DefectCategory mapping
- [ ] Configurable confidence threshold (`--min-confidence`)

### CLI
```bash
depyler oracle export-oip \
    --input-dir ./examples \
    --output ./training_corpus/citl.parquet \
    --format parquet \
    --min-confidence 0.80 \
    --include-clippy
```

### Export Schema
```json
{
  "source_file": "example.py",
  "error_code": "E0308",
  "clippy_lint": null,
  "oip_category": "TypeErrors",
  "confidence": 0.95,
  "message": "mismatched types",
  "span": {"line_start": 42, "column_start": 12},
  "suggestion": {"replacement": ".parse()", "applicability": "MaybeIncorrect"},
  "timestamp": 1732752000,
  "depyler_version": "3.21.0"
}
```

## References

- Spec: `docs/specifications/verbose-compiler-diagnostics-citl-spec.md` Section 11
- OIP ticket: organizational-intelligence-plugin NLP-014
- Data pipeline: alimentar DataLoader

## Acceptance Criteria

- [ ] Export 5K+ examples in <10 seconds
- [ ] Parquet files readable by alimentar
- [ ] All 14 error codes mapped
- [ ] All 10 clippy lints mapped
